### ISSUE TITLE
Add loss functions to train better embeddings for retrieval tasks

### ISSUE THREAD

**Issue by zheng-da** [awslabs/graphstorm#560]
Currently, link prediction is trained with cross-entropy loss that classifies the positive and negative edges. The embeddings trained with this loss may not be suitable for retrieval tasks because retrieval tasks use KNN to find similar embeddings in Euclidean space. We should use cosine similarity or dot product to compute the loss to train better embeddings for KNN search.
